src/machine/machine.c

```diff
@@ -357,28 +361,25 @@ static int machine_start_scope(
         if (r < 0)
                 return r;
 
-        r = sd_bus_message_append(m, "ss", strempty(scope), "fail");
+        r = sd_bus_message_append(m, "ss", unit, "fail");
         if (r < 0)
                 return r;
 
         r = sd_bus_message_open_container(m, 'a', "(sv)");
         if (r < 0)
                 return r;
 
-        if (!isempty(slice)) {
-                r = sd_bus_message_append(m, "(sv)", "Slice", "s", slice);
-                if (r < 0)
-                        return r;
-        }
+        r = sd_bus_message_append(m, "(sv)", "Slice", "s", SPECIAL_MACHINE_SLICE);
+        if (r < 0)
+                return r;
 
-        if (!isempty(description)) {
-                r = sd_bus_message_append(m, "(sv)", "Description", "s", description);
-                if (r < 0)
-                        return r;
-        }
+        const char *description = strjoina(machine->class == MACHINE_VM ? "Virtual Machine " : "Container ", machine->name);
```

Comment:
uhhh i don't know what this line really means: you do two stack allocations here, one via strjoina() and one via the variable declaration. Is this even defined in its behaviour? I'd really split that into two lines...

Reply:
It's like saying `const char *foo, *bar`: the compiler also does two allocations on a single line. I don't think we need to care about the details of what exactly happens there. I'm sure there are some rules that specify in which order the allocations should happen, but then the compiler is probably free to reorder them when optimizing anyway. I'm pretty sure the pointer is optimized away.

I'll make this into two lines, but I don't this matters at all. If anything, separating the declaration and initialization increases (very slightly) the chance of an error.

Reply:
still smells fishy, but ok